packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -116,3 +117,89 @@ export type TokenScope =
   | typeof AllDataReadWrite
   | Permissions
   | PredefinedScope;
+
+export interface CacheItemKey {
+  key: string;
+}
+
+export interface CacheItemKeyPrefix {
+  keyPrefix: string;
+}
+
+export function isCacheItemKey(
+  cacheItem: CacheItemKey | All
+): cacheItem is CacheItemKey {
+  return 'key' in cacheItem;
+}
+
+export function isCacheItemKeyPrefix(
+  cacheItem: CacheItemKeyPrefix | All
+): cacheItem is CacheItemKeyPrefix {
+  return 'keyPrefix' in cacheItem;
+}
+
+export type CacheItemSelector =
+  | All
+  | CacheItemKey
+  | CacheItemKeyPrefix
+  | string;
```

Comment:
you don't seem to be handling the `string` case in either sdk

Reply:
good catch @krispraws thank you, i am patching this now